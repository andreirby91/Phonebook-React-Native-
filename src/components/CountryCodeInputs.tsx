import React, {useState} from 'react';
import {View, Text, TextInput, StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  textInput: {
    height: 50,
    padding: 10,
    borderColor: 'grey',
    borderWidth: 1,
    marginTop: 10,
    fontSize: 17,
  },
  formControl: {
    marginBottom: 25,
  },
  label: {
    fontSize: 20,
  },
});

const CountryCodeInputs = ({
  onCountryUpdate,
  onCodeUpdate,
}: {
  onCountryUpdate: (country: string) => void;
  onCodeUpdate: (code: string) => void;
}) => {
  const [country, setCountry] = useState('');
  const [code, setCode] = useState('');

  return (
    <View style={{flexDirection: 'row', justifyContent: 'space-between'}}>
      <View style={{flex: 1}}>
        <Text style={styles.label}>Country:</Text>
        <TextInput
          placeholder="Type your country"
          style={[styles.formControl, styles.textInput]}
          onChangeText={(text) => {
            setCountry(text);
            onCountryUpdate(text);
          }}
          value={country}
          maxLength={30}
        />
      </View>
      <View style={{width: 150, marginLeft: 20}}>
        <Text style={styles.label}>Code:</Text>
        <TextInput
          placeholder="Type your code"
          style={[styles.formControl, styles.textInput]}
          onChangeText={(text) => {
            setCode(text);
            onCodeUpdate(text);
          }}
          value={code}
          maxLength={30}
        />
      </View>
    </View>
  );
};

export default CountryCodeInputs;
