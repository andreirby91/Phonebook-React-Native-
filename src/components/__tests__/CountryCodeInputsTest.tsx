import React from 'react';
import renderer from 'react-test-renderer';

import CountryCodeInputs from '../CountryCodeInputs';

it('renders correctly with defaults', () => {
  const button = renderer
    .create(
      <CountryCodeInputs onCodeUpdate={() => {}} onCountryUpdate={() => {}} />,
    )
    .toJSON();
  expect(button).toMatchSnapshot();
});
