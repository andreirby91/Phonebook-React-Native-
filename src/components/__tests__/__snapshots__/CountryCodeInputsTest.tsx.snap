// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with defaults 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
        }
      }
    >
      Country:
    </Text>
    <TextInput
      allowFontScaling={true}
      maxLength={30}
      onChangeText={[Function]}
      placeholder="Type your country"
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "marginBottom": 25,
          },
          Object {
            "borderColor": "grey",
            "borderWidth": 1,
            "fontSize": 17,
            "height": 50,
            "marginTop": 10,
            "padding": 10,
          },
        ]
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <View
    style={
      Object {
        "marginLeft": 20,
        "width": 150,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
        }
      }
    >
      Code:
    </Text>
    <TextInput
      allowFontScaling={true}
      maxLength={30}
      onChangeText={[Function]}
      placeholder="Type your code"
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "marginBottom": 25,
          },
          Object {
            "borderColor": "grey",
            "borderWidth": 1,
            "fontSize": 17,
            "height": 50,
            "marginTop": 10,
            "padding": 10,
          },
        ]
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
</View>
`;
